modules/warehouse/persistence/position_repository.go

```diff
@@ -89,17 +89,34 @@ func (g *GormPositionRepository) CreateOrUpdate(ctx context.Context, data *posit
 	if !ok {
 		return service.ErrNoTx
 	}
-	return tx.Save(toDBPosition(data)).Error
+	positionRow, uploadRows := toDBPosition(data)
+	if err := tx.Save(positionRow).Error; err != nil {
+		return err
+	}
+	for _, uploadRow := range uploadRows {
+		if err := tx.Save(uploadRow).Error; err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func (g *GormPositionRepository) Create(ctx context.Context, data *position.Position) error {
 	tx, ok := composables.UseTx(ctx)
 	if !ok {
 		return service.ErrNoTx
 	}
-	if err := tx.Create(toDBPosition(data)).Error; err != nil {
+	positionRow, junctionRows := toDBPosition(data)
+	if err := tx.Create(positionRow).Error; err != nil {
 		return err
 	}
+	for _, junctionRow := range junctionRows {
+		// TODO: this feels like a hack
```

Comment:
@jcbbb This feels a little off, not sure how to make it better

Reply:
I think there is a way for gorm to do it with associations, no?

Reply:
Yes, but it would save uploads also. In my case uploads already exist. 
And also doing this in a for loop feels wrong too. A bulk action approach would be great.

We may just switch to raw sql😂

Reply:
> Don't see any bulk action support in Gorm

https://gorm.io/docs/create.html#Batch-Insert

> We may just switch to raw sql😂

The sooner we learn to start projects with raw sql, the better :smile:. I would like to see your hatred towards ORMs develop more.

Reply:
I do dislike ORM's mostly. I initially started it without and ORM, but got tired of writing boiler plate, so switched to an ORM. But now thinking of going back

Reply:
That's great, I'll switch to using batch insert